feat: component dod in line with Miro (#621)
nl-design-system/documentatie: docs/componenten/definition-of-done.mdx

> 

Reviewer: Veel comments zullen gaan over het korter maken van de zinnen.

Dit component bestaat nog niet, maar de rationale is duidelijk, elke organisatie die hem nodig heeft kan hem ontwikkelen volgens NL Design System architectuur:

> - Staat in de [publieke Backlog](https://github.com/nl-design-system/backlog/issues) van NL Design System

Reviewer: ```suggestion
- Staat in de [publieke Backlog](https://github.com/nl-design-system/backlog/issues) van NL Design System.
```

> - Er is een rationale opgesteld wat het component is en waarom het nodig is

Reviewer: Zou enkel 

```suggestion
- Rationale opgesteld.
```

Voldoende zijn? Waarom? Om de d.o.d. / tasklist kort en bondig te houden zouden we ergens buiten deze lijst om kunnen uitleggen wat we met rationale bedoelen. Namelijk: Naam en doel van het component. Maar ook doel van varianten. Uiteraard beschrijven we dit in het stappenplan, maar ook ergens op de documentatie website?

> - Screenshots verzameld van het component in diverse huisstijlen

Reviewer: ```suggestion
- Screenshots verzameld voor diverse huisstijlen.
```

> - Varianten van het component die regelmatig voorkomen opgeschreven en beschreven wanneer je ze gebruikt

Reviewer: Omdat ik nu zelf bezig ben kom ik erachter dat ik niet altijd kan vinden wanneer je welke variant gebruikt. Om deze check vanuit het kernteam te kunnen halen zou ik daarom het volgende willen voorstellen.

```suggestion
- Varianten die regelmatig voorkomen zijn vastgelegd.
```

> - Screenshots verzameld van iedere variatie in diverse huisstijlen

Reviewer: Volgens mij kan deze weg omdat deze check wordt gecoverd door:
- Screenshots verzameld voor diverse huisstijlen.
- Varianten die regelmatig voorkomen zijn vastgelegd.

> - Kernteam en community verwachten dat het component voor meerdere organisaties inzetbaar kan zijn

Reviewer: Ik zie deze niet terug in onze sticky's. En omdat er ook een check is voor 'Rationale is door kernteam bevestigd' zou deze check er wat mij betreft uit kunnen. Of we moeten 'Rationale is door kernteam bevestigd' herschrijven.

> - De rationale is onderbouwd door gebruikersonderzoek (eigen of van anderen)

Reviewer: ```suggestion
- De rationale is onderbouwd door gebruikersonderzoek (eigen of van anderen).
```

> - De rationale is door het kernteam bevestigd

Reviewer: ```suggestion
- De rationale is door het kernteam bevestigd.
```

> - Component is vindbaar op de NL Design System website

Reviewer: ```suggestion
- Component is vindbaar op de NL Design System website.
```

### Community

Dit component bestaat in de community, op één of meer plekken. Om bij NL Design System als 'Community component' genoemd te worden moet het component naast alle 'Help wanted' criteria aan de volgende voorwaarden voldoen:

> - Component implementatie van 1 of meer varianten voldoen aan de rationale

Reviewer: ```suggestion
- Implementatie van 1 of meer varianten voldoen aan de rationale.
```

> - Component is publiek beschikbaar

Reviewer: ```suggestion
- Publiek beschikbaar.
```

> - Component en varianten hebben voldoende design tokens om de huisstijlen uit de verzamelde screenshots mogelijk te maken

Reviewer: ```suggestion
- Component en varianten hebben voldoende design tokens om de huisstijlen uit de verzamelde screenshots. mogelijk te maken
```

> - Component en varianten hebben voldoende design tokens om de huisstijlen uit de verzamelde screenshots mogelijk te maken

Reviewer: ```suggestion
- Varianten hebben voldoende design tokens om de huisstijlen uit de verzamelde screenshots mogelijk te maken.
```

- Component heeft de EUPL-1.2 licentie

> - Component documentatie heeft de Creative Commons 0 licentie (CC0)

Reviewer: ```suggestion
- Documentatie heeft de Creative Commons 0 licentie (CC0)
```

> - Component is beschikbaar als CSS component

Reviewer: ```suggestion
- Beschikbaar als CSS component.
```

> - Component API's zijn geprefixed met de prefix van de verantwoordelijke organisatie

Reviewer: ```suggestion
- API's hebben een prefix van de verantwoordelijke organisatie.
```

- Component staat in, publiek beschikbare, Storybook van de verantwoordelijke organisatie

> - Component is opgenomen in de [Storybook met alle NL Design System thema's](https://nl-design-system.github.io/themes/)

Reviewer: ```suggestion
- Opgenomen in de [Storybook met alle NL Design System thema's](https://nl-design-system.github.io/themes/)
```

> - Visuele regressietests zijn beschikbaar voor het component in de thema-storybook

Reviewer: ```suggestion
- Visuele regressietests zijn beschikbaar in de Thema Storybook.
```

> - Component is beschikbaar in de NL Design System Figma bibliotheek.

Reviewer: ```suggestion
- Beschikbaar in de NL Design System Figma bibliotheek.
```

> - Component storybook en figma linkjes zijn toegevoegd aan de component index

Reviewer: ```suggestion
- Storybook en Figma links zijn toegevoegd aan de component index.
```

> - Component is beschikbaar in de NLDS component Quick Scan

Reviewer: Deze file hebben we inmiddels hernoemd naar 'Component Assessment

```suggestion
- Beschikbaar in de NL Design System Component Assessment (Figma).
```

> - Component status is aangepast

Reviewer: Benoemen van de betreffende status?

```suggestion
- Status is aangepast naar Community.
```

> - Component is in een heartbeat, Slack en nieuwsbrief geïntroduceerd

Reviewer: ```suggestion
- Component is in een Heartbeat, Slack en Nieuwsbrief geïntroduceerd.
```

### Candidate

> Dit component is opgenomen in de `@nl-design-system-unstable` organisatie. Het kernteam verwacht dat het component uiteindelijk een Hall of Fame status krijgt, maar kan nog niet garanderen dat API's niet wijzigen of er onverhoopt een ander vergelijkbaar component in de plaats komt.

Reviewer: ```suggestion
Dit component is opgenomen in de `@nl-design-system-unstable` organisatie. Het kernteam verwacht dat het component uiteindelijk een Hall of Fame status krijgt, maar kan nog niet garanderen dat API's niet wijzigen of er onverhoopt een ander vergelijkbaar component voor in de plaats komt.
```

> Het component moet mogelijk nog extra goed getest worden en extra stabiel worden gemaakt.

Reviewer: ```suggestion
Het component moet mogelijk nog extra goed getest en stabiel gemaakt worden.
```

> Omdat het component wel door het kernteam onderhouden wordt voldoet het naast de meeste Community DoD eisen al wel aan veel meer criteria voor stabiel hergebruik:

Reviewer: ```suggestion
Omdat het component door het kernteam onderhouden wordt voldoet het, naast de meeste Community DoD eisen, al wel aan veel meer criteria voor stabiel hergebruik.
```

> - Community component is ingezet door 2 of meer organisaties met verschillende huisstijlen

Reviewer: ```suggestion
- Ingezet door 2 of meer organisaties met verschillende huisstijlen.
```

> - Figma component in de NL Design System bibliotheek is in lijn met het component in code

Reviewer: ```suggestion
- Component in de NL Design System Figma bibliotheek is in lijn met code.
```

> - Aliassen van niet geïmplementeerde alternatieve componenten zijn vastgelegd in documentatie

Reviewer: Volgens mij ging deze vooral om de alternatieve namen (dropdown/select). Voorstel:

```suggestion
- Aliassen van alternatieve namen voor het component zijn vastgelegd in documentatie.
```

- Anatomie van het component is vastgelegd

> - Het component is waar nodig versimpelt of opgesplitst. Zodat er één of meerdere componenten over zijn met een duidelijke use-case en documentatie

Reviewer: ```suggestion
- Component is, waar nodig, versimpelt of opgesplitst. Zodat er één of meerdere componenten over blijven met een duidelijke use-case en documentatie.
```

> - Het component heeft geen organisatie specifieke API's meer, alle API's zijn gebaseerd op een gemeenschappelijke usecase

Reviewer: ```suggestion
- Geen organisatie specifieke API's meer, alle API's zijn gebaseerd op een gemeenschappelijke use case.
```

> - API volgt conventies en is geprefixed met 'nl'

Reviewer: ```suggestion
- API volgt conventies en is geprefixed met 'nl'.
```

> - Design tokens volgen conventies en zijn geprefixed met 'nl'

Reviewer: ```suggestion
- Design tokens volgen conventies en zijn geprefixed met 'nl'.
```

- Hergebruik van logische common tokens is geïmplementeerd om theming makkelijk te houden.

> - Alle design tokens zijn algemeen nuttig voor de thema's van bestaande publieke organisaties

Reviewer: ```suggestion
- Alle design tokens zijn algemeen nuttig voor de thema's van bestaande publieke organisaties.
```

> - Component beschikbaar in NL Design System storybook

Reviewer: ```suggestion
- Beschikbaar in NL Design System storybook.
```

> - Component als candidate beschikbaar Figma library

Reviewer: ```suggestion
- Beschikbaar in NL Design System Figma bibliotheek.
```

> - Component heeft dezelfde documentatie op de website, Figma en Storybook over hoe te gebruiken

Reviewer: Ik twijfel of het overal 'dezelfde' documentatie moet zijn. Ik zou vanuit Figma voor gebruik zoveel mogelijk naar de documentatie website willen verwijzen, omdat het anders lastig up-to-date te houden is. Specifiek gebruik van een component in Figma zelf (bijvoorbeeld voor prototyping) zou natuurlijk wel in Figma horen.

Tenzij we hier eigenlijk bedoelen dat properties (type: Primary of state: Invalid etc.) in Figma en Storybook (zoveel mogelijk) gelijk moeten zijn. Maar dan zou ik ook 'properties' ipv documentatie gebruiken.

```suggestion
- Documentatie beschikbaar over gebruik.
```

Reviewer 2: ```suggestion
- Component heeft dezelfde documentatie op de website en Storybook over gebruik.
- Figma verwijst naar de documentatie op de website.
```

> - Stories, Figma voorbeelden en unit tests hebben coverage van alle features

Reviewer: ```suggestion
- Stories, Figma voorbeelden en unit tests hebben coverage van alle features.
```

> - Verwijzing naar extern gebruikersonderzoek beschikbaar die tot de keuze van dit component leiden

Reviewer: ```suggestion
- Verwijzing beschikbaar naar extern gebruikersonderzoek die tot de keuze van dit component leiden.
```

> - Accessibility en inclusie documentatie is beschikbaar voor de overwegingen die op dit component van toepassing zijn

Reviewer: ```suggestion
- Accessibility en inclusie documentatie beschikbaar voor de overwegingen van toepassing zijn.
```

> - Semantiek (roles, names, states in HTML/ARIA) van het component is goedgekeurd door het kernteam

Reviewer: ```suggestion
- Semantiek (roles, names, states in HTML/ARIA) is goedgekeurd door het kernteam.
```

> - Component implementatie en documentatie is op toegankelijkheid getest door leden van het kernteam

Reviewer: ```suggestion
- Implementatie en documentatie is op toegankelijkheid getest door leden van het kernteam.
```

> - Component is getest op geschiktheid voor internationalisatie

Reviewer: ```suggestion
- Getest op geschiktheid voor internationalisatie.
```

> - Component kan omgaan met verschillende soorten content, zoals heel veel, missende of juist heel weinig content.

Reviewer: ```suggestion
- Kan omgaan met verschillende soorten content zoals; veel, weinig of ontbrekende content.
```

Reviewer 2: ```suggestion
- Component kan omgaan met verschillende soorten content: bijvoorbeeld korte tekst, veel woorden, lange woorden, of juist ontbrekende content.
```

> - Codebase van het component gebruikt linting en automatische code-formatting met Prettier volgens NL Design System configuraties

Reviewer: ```suggestion
- Codebase gebruikt linting en automatische code-formatting met Prettier volgens NL Design System configuraties.
```

> - Component beschikbaar in NPM scope: @nl-design-system-unstable

Reviewer: ```suggestion
- Beschikbaar in NPM scope: @nl-design-system-unstable.
```

- Changelog wordt gepubliceerd bij nieuwe versies

> - Documentatie is bij de doelgroepen getest en goed bevonden

Reviewer: ```suggestion
- Documentatie is getest bij de doelgroepen en goed bevonden.
```

> - Component status is aangepast

Reviewer: ```suggestion
- Status is aangepast naar Candidate.
```

Reviewer: Bij Community was er na het aanpassen van de status nog eens stap namelijk:

- Component is in een Heartbeat, Slack en Nieuwsbrief geïntroduceerd.

Nu is introduceren misschien niet meer nodig maar het vermelden van status wisseling wellicht wel?

### Hall of fame

> Een component met deze status is bewezen door gebruikname in productie bij ten minste twee verschillende organisaties. Het component is daarmee in gebruik getest, zowel door developers als door gebruikers en heeft een toegankelijkheids audit succesvol doorlopen.

Reviewer: ```suggestion
Een component met deze status is bewezen door gebruikname in productie bij ten minste twee verschillende organisaties. Het component is daarmee in gebruik getest, zowel door developers en designers als door eindgebruikers en heeft een toegankelijkheids audit succesvol doorlopen.
```

> Het gebruiksvriendelijk en toegankelijk gebruik van het component in context is goed gedocumenteerd, zodat designers en developers in staat zijn het component op de juiste manier in de context van hun eigen project in te zetten. Ook kan je ervan uitgaan dat de API's van het component, zoals Design Tokens en naamgeving stabiel blijven en worden updates vanaf nu voorzien van duidelijke changelogs en semantische versies die `breaking changes` goed communiceren.

Reviewer: ```suggestion
Het gebruiksvriendelijk en toegankelijk gebruik van het component in context is goed gedocumenteerd, zodat designers en developers in staat zijn het component op de juiste manier in de context van hun eigen project in te zetten. Ook kan je ervan uitgaan dat de API's van het component, zoals design tokens en naamgeving stabiel blijven. Daarnaast worden updates vanaf nu voorzien van duidelijke changelogs en semantische versies die `breaking changes` goed communiceren.
```

Dit component bestaat nog niet, maar de rationale is duidelijk, elke organisatie die hem nodig heeft kan hem ontwikkelen volgens NL Design System architectuur:

- Staat in de [publieke Backlog](https://github.com/nl-design-system/backlog/issues) van NL Design System.
- Er is een rationale opgesteld wat het component is en waarom het nodig is

> - Screenshots verzameld voor diverse huisstijlen.

Reviewer 2: ```suggestion
- Screenshots verzameld voor diverse huisstijlen, van alle belangrijke varianten en states.
```

### Candidate

Dit component is opgenomen in de `@nl-design-system-unstable` organisatie. Het kernteam verwacht dat het component uiteindelijk een Hall of Fame status krijgt, maar kan nog niet garanderen dat API's niet wijzigen of er onverhoopt een ander vergelijkbaar component in de plaats komt.
Het component moet mogelijk nog extra goed getest worden en extra stabiel worden gemaakt.
Omdat het component wel door het kernteam onderhouden wordt voldoet het naast de meeste Community DoD eisen al wel aan veel meer criteria voor stabiel hergebruik:

- Community component is ingezet door 2 of meer organisaties met verschillende huisstijlen
- Figma component in de NL Design System bibliotheek is in lijn met het component in code
- Aliassen van niet geïmplementeerde alternatieve componenten zijn vastgelegd in documentatie
- Anatomie van het component is vastgelegd
- Het component is waar nodig versimpelt of opgesplitst. Zodat er één of meerdere componenten over zijn met een duidelijke use-case en documentatie
- Het component heeft geen organisatie specifieke API's meer, alle API's zijn gebaseerd op een gemeenschappelijke usecase
- API volgt conventies en is geprefixed met 'nl'

> - Design tokens volgen conventies en zijn geprefixed met 'nl'

Reviewer: We hebben na deze check een [extra sticky toegevoegd in de Miro](https://miro.com/app/board/uXjVMHZLr-c=/?moveToWidget=3458764580303967925&cot=14) namelijk: Beschikbaar in 'Default' en 'Voorbeeld' thema.

Zullen we die dan hier ook toevoegen? En zo ja, dan moeten we deze ook aan het Candidate board toevoegen.

Dit component bestaat nog niet, maar de rationale is duidelijk, elke organisatie die hem nodig heeft kan hem ontwikkelen volgens NL Design System architectuur:

- Staat in de [publieke Backlog](https://github.com/nl-design-system/backlog/issues) van NL Design System.
- Rationale opgesteld voor naam en doel.
- Screenshots verzameld voor diverse huisstijlen, van alle belangrijke varianten en states.
- Varianten van het component die regelmatig voorkomen opgeschreven en beschreven wanneer je ze gebruikt

> - Screenshots verzameld van iedere variatie in diverse huisstijlen

Author: @Robbert deze bedoel ik
